control/controlclient/debug.go

```diff
@@ -42,28 +43,51 @@ func dumpGoroutinesToURL(c *http.Client, targetURL string) {
 	}
 }
 
-var reHexArgs = regexp.MustCompile(`\b0x[0-9a-f]+\b`)
+func isHexString(s string) bool {
+	if len(s) <= 2 || s[0:2] != "0x" {
+		return false
+	}
+	_, err := strconv.ParseUint(s[2:], 16, 64)
```

Comment:
One thing that worries me about this is that it may overflow if the hex string is big enough. Does that matter? I'm not sure if opting for a for loop which checks every character would be better, but it'd be more explicit. I am also curious whether capital hex such as 0xFF is valid, which wasn't previously accepted?

Reply:
Added a test for capitalized hex numbers, even if they were allowed before we want to suppress them.

As it is a backtrace from the go runtime, I don't think it can be bigger than 64 bits. We don't run on machines with wider integers than that.

If a very wide hex number appears, the worst that happens is we don't suppress it.